cbmc_viewer/metric_reportt.py

```diff
@@ -0,0 +1,392 @@
+    'count': int,
+    'link': str
+    },required=True)
+
+VALID_ARRAY_CONSTRAINT_STATS = voluptuous.schema_builder.Schema({
+    'count': int,
+    'link': str
+    },required=True)
+
+VALID_CBMC_PROOF_STATS = voluptuous.schema_builder.Schema({
+    'programSteps': int,
+    'vccCount': int,
+    'varCount': int,
+    'clauseCount': int
+    },required=True)
+
+VALID_CLAUSE_STATS = voluptuous.schema_builder.Schema({
+    'clauseCoreCount': int,
+    'link': str
+    },required=True)
+
+VALID_RUNTIME_OBJ = voluptuous.schema_builder.Schema({
+    'val': float,
+    'string': str
+    },required=True)
+
+VALID_RUNTIME = voluptuous.schema_builder.Schema({
+    'symex': VALID_RUNTIME_OBJ,
+    'convertSSA': VALID_RUNTIME_OBJ,
+    'postProc': VALID_RUNTIME_OBJ,
+    'solver': VALID_RUNTIME_OBJ,
+    'maxKey': {
+        'val': str,
+        'string': str
+        }
+    },required=True)
+
+VALID_RUNTIME_ANALYSIS_SUMMARY = voluptuous.schema_builder.Schema([{
+    'jobname': str,
+    'link': str,
+    'byteop': VALID_BYTEOP_STATS,
+    'alias': VALID_ALIAS_STATS,
+    'memop': VALID_MEMOP_STATS,
+    'array': VALID_ARRAY_CONSTRAINT_STATS,
+    'cbmc': VALID_CBMC_PROOF_STATS,
+    'clause': VALID_CLAUSE_STATS,
+    'runtimeKissat': float,
+    'runtime': VALID_RUNTIME
+    }],required=True)
+
+VALID_SUMMARY = voluptuous.schema_builder.Schema({
+    'summary': VALID_RUNTIME_ANALYSIS_SUMMARY
+    },required=True)
+
+################################################################
+
+class RuntimeAnalysisSummary:
+
+    def __init__(self, proofs, srcdir):
+        self.summary = []
+
+        for proof in proofs:
+            jobname = proof
+            proof_path = os.path.abspath(proof)
+            outdir = os.path.join(proof_path, 'html', 'html')
+
+            byteop = os.path.join(proof_path, 'html', 'json', 'viewer-byteop.json')
+            alias = os.path.join(proof_path, 'html', 'json', 'viewer-alias.json')
+            memop = os.path.join(proof_path, 'html', 'json', 'viewer-memop.json')
+            array = os.path.join(proof_path, 'html', 'json', 'viewer-array.json')
+            result = os.path.join(proof_path, 'html', 'json', 'viewer-result.json')
+            clause = os.path.join(proof_path, 'html', 'json', 'viewer-clause.json')
+            kissat = os.path.join(proof_path, 'logs', 'kissat.txt')
+            
+            self.summary.append({
+                'jobname': jobname,
+                'link': get_link_to_job(outdir),
+                'byteop': get_byteop_stats(byteop, outdir),
+                'alias': get_alias_stats(alias, outdir),
+                'memop': get_memop_stats(memop, outdir),
+                'array': get_array_stats(array, outdir),
+                'cbmc': get_cbmc_proof_stats(result),
+                'clause': get_clause_stats(clause, outdir),
+                'runtimeKissat': get_kissat_runtime(kissat),
+                'runtime': get_cbmc_runtime_stats(result)
+                })
```

Comment:
Is this just reading the json serialization of the various objects you have created?